src/frontend/src/lib/components/views/AuthorizeNewDevice.svelte

```diff
@@ -40,24 +59,39 @@
     <br /><br />
     <b class="text-text-primary">Never</b> enter a code from another source.
   </p>
-  <Input
+  <CodeInput
+    bind:element={inputRef}
     bind:value={confirmationCode}
-    placeholder="XXXXXX"
-    size="md"
-    class="mb-8"
+    length={CODE_LENGTH}
+    class="mb-3"
+    error={isInvalidCode
+      ? "Invalid code. Please check and try again."
+      : undefined}
+    hint={"\u00a0"}
   />
   <Button
     onclick={handleSubmit}
     variant="primary"
     size="xl"
     type="submit"
-    disabled={confirmationCode.length === 0 || loading}
+    disabled={confirmationCode.length < CODE_LENGTH || isConfirming}
     class="mb-3"
```

Comment:
ditto

Reply:
I meant more to have "props" to change outer space. Outer space should not be the responsibility of the component (with or without props) but always of the parent.

Reply:
Using flex or margins it's ok for me. It's where this is handled that I mean.